vars/buildPlugin.groovy

```diff
@@ -282,6 +281,13 @@ def call(Map params = [:]) {
   }
 }
 
+private void discoverReferenceBuild() {
+  folders = env.JOB_NAME.split('/')
+  if (folders.length > 1) {
+    discoverGitReferenceBuild(scm: folders[1])
```

Comment:
What is this doing? If `JOB_NAME` were e.g. `some/path/to/some/folder/and/then/your/job-name/PR-123`, you are computing `path`??

Perhaps you meant `folders[-2]` (`job-name` in this example)? Or `../master`? Or 

[Docs for the step](https://www.jenkins.io/doc/pipeline/steps/git-forensics/#discovergitreferencebuild-discover-git-reference-build) list the `scm` parameter but it has no documentation.

Reply:
In the `discoverGitReferenceBuild` step the reference build is discovered from the associated SCM. In order to find the correct SCM (there are multiple ones available and Jenkins sadly does not define a *main* SCM) the ID `git https://github.com/jenkinsci/analysis-model.git'` is matched against the `SCM` parameter (it is documented in the JavaDoc and UI). Seems that we need a third place were we would copy and paste that documentation right now (`help-scm.html`). 

I am using the job name to identify the correct SCM. So for plugins we do not have a generic `some/path/to/some/folder/and/then/your/job-name/PR-123`, we always have the fixed name `Plugins/job-name/PR-738`. So I am using folders[1] to extract the `job-name`. But `[-2]` as index should work as well and is more stable when we might refactor the jobs structure in the future.

Reply:
> there are multiple ones available and Jenkins sadly does not define a _main_ SCM

For branch project it does. https://javadoc.jenkins.io/plugin/scm-api/jenkins/scm/api/SCMRevisionAction.html

> more stable when we might refactor the jobs structure

Exactly.